src/containers/ActivityContainer/FilterModal.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Image from 'next/image';
+
+import CButton from 'src/components/CButton';
+import { useAppSelector } from 'src/hooks/useRedux';
+import { IToken } from 'src/reducers/tokens';
+
+import close from 'public/images/whiteClose.svg';
+import blackClose from 'public/images/close.svg';
+import searchLogo from 'public/images/search.svg';
+import { IFilterTokens } from 'src/constants/types';
+import defaultToken from 'public/images/defaultToken.svg';
+import CBottomSheet from 'src/components/CBottomSheet';
+import CInput from 'src/components/CInput';
+
+type ModalProps = {
+  open: boolean;
+  closeModal: () => void;
+  initialSentChecked: boolean;
+  selectedTokenValue: IToken[];
+  setIsOpen: (_: boolean) => void;
+  initialReceivedChecked: boolean;
+  setSelectedTokenValue: React.Dispatch<React.SetStateAction<IToken[]>>;
+  handleSubmitFilter: (filteredValuesObject: IFilterTokens, selectedTokens: IToken[]) => void;
+};
+
+const FilterModal = ({
+  open,
+  setIsOpen,
+  closeModal,
+  handleSubmitFilter,
+  selectedTokenValue,
+  setSelectedTokenValue,
+  initialReceivedChecked,
+  initialSentChecked,
+}: ModalProps) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [receivedChecked, setReceivedChecked] = useState(true);
+  const [sentChecked, setSentChecked] = useState(true);
+
+  const tokens = useAppSelector((store) => store.tokens);
+  const modalRef = useRef<HTMLDivElement | null>(null);
+  const bottomSheetRef = useRef<HTMLDivElement | null>(null);
+  const secondModalRef = useRef<HTMLUListElement | null>(null);
```

Comment:
Create a type in models.ts called Optional<T> and use it anywhere that uses T | null as a type. 

For example

`  const bottomSheetRef = useRef<Optional<HTMLDivElement>>(null);`